packages/@uppy/webdav/src/Webdav.tsx

```diff
@@ -0,0 +1,117 @@
+import { h } from 'preact'
+import { useCallback, useState } from 'preact/hooks'
+
+import { UIPlugin } from '@uppy/core'
+import { Provider, tokenStorage } from '@uppy/companion-client'
+import { ProviderViews } from '@uppy/provider-views'
+
+import packageJson from '../package.json'
+import locale from './locale.ts'
+
+class WebdavSimpleAuthProvider extends Provider {
+  async login({ authFormData, uppyVersions, signal }) {
+    return this.loginSimpleAuth({ uppyVersions, authFormData, signal })
+  }
+
+  async logout() {
+    this.removeAuthToken()
+    return { ok: true, revoked: true }
+  }
+}
+
+const AuthForm = ({ loading, i18n, onAuth }) => {
+  const [webdavUrl, setWebdavUrl] = useState('')
+
+  const onSubmit = useCallback(
+    (e) => {
+      e.preventDefault()
+      onAuth({ webdavUrl: webdavUrl.trim() })
+    },
+    [onAuth, webdavUrl],
+  )
```

Comment:
We don't need a `useCallback` for this. Note that formatting will probably complain.


```suggestion
  const onSubmit = (event) => {
      event.preventDefault()
      onAuth({ webdavUrl: webdavUrl.trim() })
   }
```

Reply:
oops i just realised you removed this. why is it not needed? generally it's considered best practice to useCallback for props given to react components, to prevent re-rendering the whole subtree on every render.